### PR TITLE
REGR: ignore empty geometries in plot

### LINKED ISSUE #1572: BUG: Plotting empty geometries generates a Type Error (shapely only)
I'm not sure if this is a bug or expected behaviour. I've tested this in geopandas 0.7 and 0.8.1.

I have a process which generates some geometry but it may fail. I use that geometry for further operations so it seems prefereable to keep the failed geometries as empty geometries (e.g. Polygon()) rathern than missing (None). However when I plot the final geodataframe the plot will fail with a TypeError if the gemetries are empty but plot as expected if they are missing.

The plotting issue is easily reproducible with the [Creating a GeoDataFrame from a DataFrame with coordinates example](https://geopandas.org/gallery/create_geopandas_from_pandas.html). I would expect it to plot the same as the version with None.

```
import pandas as pd
import geopandas as gpd
from shapely.geometry import Point

df = pd.DataFrame(
    {'City': ['Buenos Aires', 'Brasilia', 'Santiago', 'Bogota', 'Caracas'],
     'Country': ['Argentina', 'Brazil', 'Chile', 'Colombia', 'Venezuela'],
     'Latitude': [-34.58, -15.78, -33.45, 4.60, 10.48],
     'Longitude': [-58.66, -47.91, -70.66, -74.08, -66.86]})

gdf = gpd.GeoDataFrame(
    df, geometry=gpd.points_from_xy(df.Longitude, df.Latitude))

gdf.plot(color='red') # plots okay
```
Missing geometry plots okay
```
gdf.loc[3,'geometry'] = None
gdf.plot(color='red') # plots okay
```
Empty geometry raises a TypeError
```
gdf.loc[3,'geometry'] = Point()
gdf.plot(color='red') # raises a TypeError
```

### PR DESCRIPTION
**Pull request by martinfleis**
Fixes #1572

Part of the original issue was fixed in #1677, the rest is here.

### CODE REVIEW

File: geopandas/plotting.py [hunk 1]
```diff
@@ -283,8 +283,8 @@ def _plot_point_collection(
     geoms, multiindex = _flatten_multi_geoms(geoms)
     # values are expanded below as kwargs["c"]
 
-    x = [p.x for p in geoms]
-    y = [p.y for p in geoms]
+    x = [p.x if not p.is_empty else None for p in geoms]
+    y = [p.y if not p.is_empty else None for p in geoms]
```
Comment (jorisvandenbossche) [inline geopandas/plotting.py:287]: Do we know here we already can't have None?

Comment (jorisvandenbossche) [reply geopandas/plotting.py:287]: Ah, yes, because at a higher level we explicitly selected the points

File: geopandas/tests/test_plotting.py [hunk 2]
```diff
@@ -308,6 +308,19 @@ def test_empty_geometry(self):
             ax = s.plot()
             assert len(ax.collections) == 1
 
+        # more complex case with GEOMETRYCOLLECTION EMPTY, POINT EMPTY and NONE
+        poly = Polygon([(-1, -1), (-1, 2), (2, 2), (2, -1), (-1, -1)])
+        point = Point(0, 1)
+        point_ = Point(10, 10)
+        empty_point = Point()
+        none_point = Point()
```
Comment (jorisvandenbossche) [inline geopandas/tests/test_plotting.py:316]: This is the same as the line above, so can be removed I think?

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for looking at this!

File: geopandas/plotting.py (hunk 1 above)
Comment (martinfleis) [reply geopandas/plotting.py:287]: This only covers empty point cases, so we have Point object but it has no coordinates to return and errors.

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**